main/reference/vstorage-ref.md

```diff
@@ -1,13 +1,128 @@
-# VStorage Reference
+# VStorage
 
-See also:
+In the Agoric platform, VStorage is a key-value store that:
 
-- [Querying VStorage](/guides/getting-started/contract-rpc#querying-vstorage)
-- [Publishing to chainStorage](../guides/zoe/pub-to-storage)
-- [x/vstorage module](https://github.com/Agoric/agoric-sdk/tree/003f0c2232815a8d64a3f9a5b05521a10160ce34/golang/cosmos/x/vstorage#readme)
+- provides a **read-only interface where clients of the consensus layer** can query the stored data using specific paths.
+- is organized in a **hierarchical, path-based structure**. Clients can query these paths to retrieve the data. For example:
+```sh
+$ agd query vstorage keys 'published.vaultFactory.managers.manager0.vaults'
+children:
+- vault0
+```
+- can be written through a specialized API called **[chainStorage](https://docs.agoric.com/guides/zoe/pub-to-storage.html#publishing-to-chainstorage)** from within the JavaScript VM.
+
+![vstorage query diagram](../guides/getting-started/assets/vstorage-brand-q.svg)
```

Comment:
This page now has a lot of overlap with `/guides/getting-started/contract-rpc.html`, so there should be links in both directions to clarify how they relate. Perhaps reduce the vstorage material in `contract-rpc`?

Reply:
The idea is to create a centralized reference for VStorage. I feel centralized reference is okay like this without reducing context at some place else.

Reply:
Expanding the scope of this page from [reference](https://docs.divio.com/documentation-system/reference/) to also cover [explanation](https://docs.divio.com/documentation-system/explanation/)/[how-to](https://docs.divio.com/documentation-system/how-to-guides/) seems awkward, especially since it overlaps with several other discussions of vstorage.